### ISSUE TITLE
#tag feature

### ISSUE THREAD

**Issue by AdrianoFerrari** [gingko/client#148]
From #147 : 
> Perhaps the syntax #tag could be displayed as a link to the search function with 'tag' as search term. That means, say I have I tag #to_do, if I click it, it searches for all of my cards with a to_do hashtag. Doing the same with the '@' symbol would be handy for using it with names. For both of them auto-complete would be lovely.

**Comment by brainchild0** [comment]
I'm not sure I understand what is suggested.  It seems as though   it might be that typing a hash symbol suspends normal editing and invokes a search feature, but I suspect a different meaning is intended.

**Comment by AdrianoFerrari** [comment]
Essentially, this is the way it works on the online version. Step 1, we render anything like "#tag" differently when cards are saved, and clicking on those adds the search term "#tag" to the search window (thereby filtering the tree to only those containing that text).

Step 2 in the feature would be to add things like autocomplete inline (while editing a card), like this:
![Autocomplete-List-With-Custom-Trigger-AutoSuggest js](https://user-images.githubusercontent.com/323678/68550355-13e3d280-03d0-11ea-84a8-a507273b19e4.png)
both in the search area and while typing.

**Comment by brainchild0** [comment]
I see.  Tagging is useful and featured in many contexts.  

In the current case, the input format is MarkDown, leading to a few observations:

- Hash symbols for tags, while widely recognized, barely avoids colliding with ATX-style MarkDown headers (e.g. `# Chapter 1`), so a different representation might be appropriate in this context.
- Any text embedded into the body of the card would seem to be part of the content. It would be hard to find general solution for giving desired treatment to tags designated in such way that fully averts  unwanted effects.

Perhaps a more robust solution would be a representation in which the tag set for each card is represented as metadata, displayed and persisted outside of the body text of the card.  A keyboard shortcut could easily invoke a mode for entering a name of a new tag.  Moreover, display of all the tags in one place makes it easier to review and to modify them, while display inside the card might disrupt the flow of reading.

**Comment by AdrianoFerrari** [comment]
I agree with your points. But I also have to find ways that, while not ideal, are at least achievable in the short term.

I am still just one person contributing to the project regularly, and have to pick where to focus my time and energy carefully, otherwise it gets overwhelming. If we can have 80% of the benefits of tags, with 20% of the time investment, sometimes I'll have to make that suboptimal choice.

That said, I do appreciate detailed thoughts like this on feature design. Sometimes it'll lead to a way of developing a feature that's both easier *and* better.

**Comment by brainchild0** [comment]
Understand. 

 I wasn't trying to be nit-picky as much as to prompt discussion to help prioritize features.  My idea was that unhelpful suggestions could simply be postponed or dropped without causing any damage. 

But I accept that the comments might target an inappropriate level of detail for the current stages of development.
